js/base-chat.js

```diff
@@ -713,6 +725,29 @@ function baseChat () {
 		if (!params[2]) {
 			d20plus.chat.resetSendMyself();
 		}
+		const macroJS = d20plus.cfg.getOrDefault("chat", "executeJSMacro");
+
+		if (macroJS !== "none") {
+			const template = /#(?<macroid>[^ ^#]+)/g;
+			params[0] = params[0].replace(template, (...string) => {
+				const macroId = string.last().macroid;
+				const macroObj = d20plus.ut.getMacroByName(macroId);
+				if (!macroObj) return string[0];
+				const macro = macroObj.attributes.action;
+				const script = d20plus.engine.decodeScript(macro);
+				if (!script) return string[0];
+				if (macroObj.collection.player.id !== d20_player_id && macroJS !== "all") {
+					d20plus.ut.sendHackerChat(`
+						Enable execution for scripts shared by other players
+						(select Execute All in betteR20 options for JS Script).
+						You should do this only if you trust your GM
+					`, true);
+					return "";
+				}
+				return d20plus.engine.runScript(script);
```

Comment:
1: is this `.replace`d string used/posted to chat..? 

2: Do we want to, say, handle `async` macros?

Reply:
1. Yes
2. No (one of the reasons being 1.)

In cases like input "Hello my name is #tokenName" (if it's a roll20 macro or a js macro with return statement) #macro call will be replaced and posted to chat. Which is obviously impossible with async.

Ofc we can try to circumvent this (I did similar things for chatQoL) but here it's not needed IMO. No need to complicate things. If anyone would want some async functionality inside a macro - one can easily write anything, including async calls and output to chat afterwards.

Reply:
fair enough, fine by me